grub-core/lib/gpt.c

```diff
@@ -373,6 +378,11 @@ grub_gpt_check_backup (grub_gpt_t gpt)
   if (backup <= end)
     return grub_error (GRUB_ERR_BAD_PART_TABLE, "invalid backup GPT LBA");
 
+  /* If both primary and backup are valid but differ prefer the primary.  */
```

Comment:
It still applies. It is rejecting any backup header that doesn't agree with a valid primary header.